### PR TITLE
doc/md clarify ent enum usage

### PR DESCRIPTION
**Pull request by yonidavidson**
Explain enum types in ent.

### CODE REVIEW

**Comment by yonidavidson** [comment]
Fix #1864

File: doc/md/schema-fields.md [hunk 1]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
 }
 ```
 
+## Enum Fields
+
+Use Ent API to generate Enum type and define possible values:
+```go
```
Comment (a8m) [inline doc/md/schema-fields.md:619]: My rephrase this or use this:

```suggestion
The `Enum` builder allows creating enum fields with a list of permitted values. 

```go
```

File: doc/md/schema-fields.md [hunk 2]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
...
+				"big",
+				"small",
+			),
+	}
+}
+```
+
+When custom type that is convertible to the Go basic type:
+
+```go
```
Comment (a8m) [inline doc/md/schema-fields.md:636]: Merge the 2 code blocks to one. Easier to follow.

File: doc/md/schema-fields.md [hunk 3]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
...
+		field.Enum("size").
+			Values(
+				"big",
+				"small",
+			),
+	}
+}
+```
+
+When custom type that is convertible to the Go basic type:
```
Comment (a8m) [inline doc/md/schema-fields.md:634]:
```suggestion
When a custom [`GoType`](#go-type) is being used, it is must be convertible to the basic `string` type or it needs to implement the [ValueScanner](https://pkg.go.dev/entgo.io/ent/schema/field#ValueScanner) interface. 

The [EnumValues](https://pkg.go.dev/entgo.io/ent/schema/field#EnumValues) interface is also required by the custom Go type to tell Ent what are the permitted values of the enum. 

The following example shows how to define an `Enum` field with a custom Go type that is convertible to `string`: 
```

File: doc/md/schema-fields.md [hunk 4]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
...
+func (Shape) Values() (roles []string) {
+	for _, r := range []Shape{Triangle, Circle} {
+		roles = append(roles, string(r))
+	}
+	return
+}
+
+```
+When custom type is not convertible to the Go basic type:
+
```
Comment (a8m) [inline doc/md/schema-fields.md:670]:
```suggestion
The following example shows how to define an `Enum` field with a custom Go type that is not convertible to `string`, but it implements the [ValueScanner](https://pkg.go.dev/entgo.io/ent/schema/field#ValueScanner) interface: 

```

File: doc/md/schema-fields.md [hunk 5]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
...
+// Fields of the User.
+func (User) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("first_name"),
+		field.String("last_name"),
+		field.Enum("size").
+			Values(
+				"big",
+				"small",
+			),
```
Comment (a8m) [inline doc/md/schema-fields.md:753]:
```suggestion
			Values("big", "small"),
```

File: doc/md/schema-fields.md [hunk 6]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
...
+		// A convertible type to string.
+		field.Enum("shape").
+			GoType(property.Shape("")),
+		// Add conversion to and from string
```
Comment (a8m) [inline doc/md/schema-fields.md:757]:
```suggestion
		// Add conversion to and from string.
```

File: doc/md/schema-fields.md [hunk 7]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
...
+client.User.Create().
+		SetFirstName("John").
+		SetLastName("Dow").
+		SetSize(user.SizeSmall).
+		SetShape(property.Triangle).
+		SetLevel(property.Low).
+		SaveX(context.Background())
+
+	john := client.User.Query().FirstX(context.Background())
+	fmt.Println(john) // User(id=1, first_name=John, last_name=Dow, size=small, shape=TRIANGLE, level=LOW)
```
Comment (a8m) [inline doc/md/schema-fields.md:775]:
```suggestion
client.User.Create().
	SetFirstName("John").
	SetLastName("Dow").
	SetSize(user.SizeSmall).
	SetShape(property.Triangle).
	SetLevel(property.Low).
	SaveX(context.Background())
	
	john := client.User.Query().FirstX(context.Background())
	fmt.Println(john)
	// User(id=1, first_name=John, last_name=Dow, size=small, shape=TRIANGLE, level=LOW)
```

**Review by a8m — APPROVED** [review]
Thanks for working on this @yonidavidson. 

Please fix these minor comments and I'll merge it 🙏

File: doc/md/schema-fields.md [hunk 8]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
...
+	Circle   Shape = "CIRCLE"
+)
+
+// Values provides list valid values for Enum.
+func (Shape) Values() (roles []string) {
+	for _, r := range []Shape{Triangle, Circle} {
+		roles = append(roles, string(r))
+	}
+	return
+}
```
Comment (giautm) [inline doc/md/schema-fields.md:666]:
```suggestion
func (Shape) Values() (kinds []string) {
	for _, r := range []Shape{Triangle, Circle} {
		kinds = append(kinds, string(r))
	}
	return
}
```

File: doc/md/schema-fields.md (hunk 2 above)
Comment (yonidavidson) [reply doc/md/schema-fields.md:638]: I actually like the split, looks more like a recipe and easier for the reader to understand what is "in ent" and what Is part of my own type (outside ent).

File: doc/md/schema-fields.md [hunk 9]
```diff
@@ -613,6 +613,168 @@ func (User) Fields() []ent.Field {
...
+	SetFirstName("John").
+	SetLastName("Dow").
+	SetSize(user.SizeSmall).
+	SetShape(property.Triangle).
+	SetLevel(property.Low).
+	SaveX(context.Background())
+	
+	john := client.User.Query().FirstX(context.Background())
+	fmt.Println(john)
+	// User(id=1, first_name=John, last_name=Dow, size=small, shape=TRIANGLE, level=LOW)
```
Comment (a8m) [inline doc/md/schema-fields.md:775]:
```suggestion
john := client.User.Query().FirstX(context.Background())
fmt.Println(john)
// User(id=1, first_name=John, last_name=Dow, size=small, shape=TRIANGLE, level=LOW)
```

**Review by a8m — APPROVED** [review]
Thanks for your contribution @yonidavidson, and thanks @giautm for the review 🚀 🙏

**Pull request merged**